lib/metadataTypes/Query.js

```diff
@@ -50,6 +50,32 @@ class Query extends MetadataType {
             key
         );
     }
+    /**
+     * a function to start query execution via API
+     * @param {string[]} keyArr customerkey of the metadata
+     */
+    static async execute(keyArr) {
+        if (keyArr[0] == null) {
+            keyArr = await this.findItemsToExecute();
+        }
+        for (const key of keyArr) {
+            const uri = `/automation/v1/queries/${key}/actions/start/`;
+            Util.logger.info(`Starting item ${key}`);
+            super.executeREST(uri);
+        }
+        return 'OK';
```

Comment:
- [ ] replace static return with `Promise.all()` logic

example from `TriggeredSend.js > refresh()`:

```js
        const refreshList = [];
        Util.logger.info(`Refreshing ${keyArr.length} ${this.definition.typeName}...`);
        Util.logger.debug(`Refreshing keys: ${keyArr.join(', ')}`);
        for (const key of keyArr) {
            refreshList.push(this._refreshItem(key, checkKey));
        }
        const successCounter = (await Promise.all(refreshList)).filter(Boolean).length;
        Util.logger.info(`Refreshed ${successCounter} of ${keyArr.length}`);
        return successCounter === keyArr.length;

```

I appended the return-line which does not exist in the refresh function. That was overlooked when designing the refresh function and should also be added there in a future release. Alternatively, one could imagine to return for example an object with success & error counters or even a more detailed object that clearly lists which ones worked and which ones didn't... given that we don't have such a requirement, a simple "all worked" or "at least one failed" solution like the suggested one above will likely do.